executor/builder.go

```diff
@@ -1119,16 +1125,23 @@ func (b *executorBuilder) buildTopN(v *plan.PhysicalTopN) Executor {
 		b.err = errors.Trace(b.err)
 		return nil
 	}
+	var initChunkSize int
+	if v.Count < uint64(b.ctx.GetSessionVars().MaxChunkSize) {
+		initChunkSize = int(v.Count)
+	} else {
+		initChunkSize = b.ctx.GetSessionVars().MaxChunkSize
+	}
 	sortExec := SortExec{
-		baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ExplainID(), childExec),
+		baseExecutor: newBaseExecutorWithInitChunkSize(b.ctx, v.Schema(), v.ExplainID(), initChunkSize, childExec),
 		ByItems:      v.ByItems,
 		schema:       v.Schema(),
 	}
 	metrics.ExecutorCounter.WithLabelValues("TopNExec").Inc()
-	return &TopNExec{
+	e := &TopNExec{
```

Comment:
Why change this?

Reply:
ok, that isn't no need after address @zhexuany's comment